### ISSUE TITLE
All splits have infinite score with MultiRMSE

### ISSUE THREAD

**Issue by wht0948** [catboost/catboost#2557]
Problem: 
**When I used MultiRMSE to train the model, the following error occurred**

_from catboost import Pool, CatBoostRegressor
train_pool = Pool(data[:,:-2], data[:,-2:])
params={
    'loss_function': 'MultiRMSE',
    'eval_metric': 'MultiRMSE',
    'iterations': 3000,
    'learning_rate': 0.03,
    'depth': 6,
    'min_data_in_leaf': 500,
    'l2_leaf_reg': 20,
    'task_type':'GPU',
    'devices':'0:1:2:3:4:5:6:7'
}
model = CatBoostRegressor(**params)
model.fit(train_pool)_

CatBoostError: /src/catboost/catboost/cuda/methods/greedy_subsets_searcher/greedy_search_helper.cpp:539: All splits have infinite score. Probably, numerical overflow occurs in loss function and/or split score calculation. Try increasing l2_leaf_reg, and/or decreasing learning_rate, etc.

catboost version: 1.2.2
Operating System: Ubuntu 20.04.5 LTS
CPU: Intel(R) Xeon(R) Processor @ 2.90GHz
GPU: A100*8

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
@wht0948 this means that your dataset is likely to contains singularities, e.g. very large labels
did you check the label column and/or try to increase l2_leaf_reg as exception message suggests?

**Comment by wht0948** [comment]
> @wht0948 this means that your dataset is likely to contains singularities, e.g. very large labels did you check the label column and/or try to increase l2_leaf_reg as exception message suggests?

Yes, I tried it but it didn't work, and when I use CPU training everything is normal

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
unlike on gpu, catboost computes with double precision on cpu, and this may be the reason

to debug this issue, we need an end-to-end, self-contained reproducer

**Comment by wht0948** [comment]
> unlike on gpu, catboost computes with double precision on cpu, and this may be the reason
> 
> to debug this issue, we need an end-to-end, self-contained reproducer

You can use this code to reproduce. Currently, it seems that when the number of data is very large, an error will be reported.

import numpy as np
from catboost import Pool, CatBoostRegressor, sum_models
data = np.random.random((100000000, 323))
data= np.array(data, order='F', dtype=np.float32)

params={
    'loss_function': 'MultiRMSE',
    'eval_metric': 'MultiRMSE', 
    'iterations': 5000, 
    'learning_rate': 0.05,
    'depth': 6,
    'random_seed': 123, 
    'min_data_in_leaf': 500, 
    'l2_leaf_reg': 5,
    'metric_period': 1,
    'task_type':'GPU',
    'devices':'0:1:2:3:4:5:6:7'
}
model = CatBoostRegressor(**params)
model.fit(data[:,:-6], data[:,[-1,-2]])
